### PR TITLE
Add Header content control template for Fluent and Simple themes

### PR DESCRIPTION
**Pull request by thevortexcloud**
## What does the pull request do?

The `HeaderedContentControl` does not currently have any template. This adds one to both the Simple and Fluent themes. As was discussed in https://github.com/AvaloniaUI/Avalonia/discussions/7875

It also adds a new page to the Control Catalogue with some examples showing how to use `HeaderedContentControl`.

## What is the current behavior?

Currently there is no template at all for `HeaderedContentControl`, which makes any attempt to use it render non-tangible content.

## What is the updated/expected behavior with this PR?

The control should now show a header and content according to their templates and header/content values when using the Simple or Fluent themes.

## How was the solution implemented (if it's not obvious)?

A new template was added to the Simple and Fluent themes.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

## Breaking changes

None known. But it's possible it may cause some issues with people who have themed the control themselves previously.

## Fixed issues

Fixes https://github.com/AvaloniaUI/Avalonia/discussions/7875

### CODE REVIEW

File: src/Avalonia.Themes.Fluent/Controls/HeaderedContentControl.xaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    <Setter Property="Padding" Value="3" />
+    <Setter Property="Template">
+      <ControlTemplate>
+        <Border Name="PART_Border"
+                BorderBrush="{TemplateBinding BorderBrush}"
+                BorderThickness="{TemplateBinding BorderThickness}"
+                CornerRadius="{TemplateBinding CornerRadius}"
+                Background="{TemplateBinding Background}"
+                Padding="{TemplateBinding Padding}">
+          <StackPanel Orientation="Vertical">
```
Comment (maxkatz6) [inline src/Avalonia.Themes.Fluent/Controls/HeaderedContentControl.xaml:16]: Wouldn't it make more sense if it was a DockPanel? Or Grid?
Where PART_HeaderPresenter has DockPanel.Dock="Top"

Comment (maxkatz6) [reply src/Avalonia.Themes.Fluent/Controls/HeaderedContentControl.xaml:16]: This way content won't have infinite height by default, making it usable with list controls

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0040016-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0040020-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

File: src/Avalonia.Themes.Fluent/Controls/HeaderedContentControl.xaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <Setter Property="Template">
+      <ControlTemplate>
+        <Border Name="PART_Border"
+                BorderBrush="{TemplateBinding BorderBrush}"
+                BorderThickness="{TemplateBinding BorderThickness}"
+                CornerRadius="{TemplateBinding CornerRadius}"
+                Background="{TemplateBinding Background}"
+                Padding="{TemplateBinding Padding}">
+          <Grid RowDefinitions="auto *">
+            <ContentPresenter Name="PART_HeaderPresenter"
```
Comment (robloo) [inline src/Avalonia.Themes.Fluent/Controls/HeaderedContentControl.xaml:17]: This is correctly a template part. It's used in code-behind here:

https://github.com/AvaloniaUI/Avalonia/blob/b2f2c49a19979946a5315dbce000f74baec0892a/src/Avalonia.Controls/Primitives/HeaderedContentControl.cs#L64C58-L68

However, that file is missing the template part attribute. Could you please modify it at the top to include:

`[TemplatePart("PART_HeaderPresenter", typeof(ContentPresenter))]`

See here for reference:

https://github.com/AvaloniaUI/Avalonia/blob/b2f2c49a19979946a5315dbce000f74baec0892a/src/Avalonia.Controls/ContentControl.cs#L16C1-L16C1

Note that the below "PART_ContentPresenter" is correct already and the template part attribute is inherited.

Comment (robloo) [reply src/Avalonia.Themes.Fluent/Controls/HeaderedContentControl.xaml:16]: Looks good, thanks. I can't mark these as resolved so if you want them to collapse go ahead and do so.

File: src/Avalonia.Themes.Fluent/Controls/HeaderedContentControl.xaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<ResourceDictionary xmlns="https://github.com/avaloniaui"
+                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+                    x:ClassModifier="internal">
+  <ControlTheme x:Key="{x:Type HeaderedContentControl}"
+                TargetType="HeaderedContentControl">
+
+    <Setter Property="Padding" Value="3" />
+    <Setter Property="Template">
+      <ControlTemplate>
+        <Border Name="PART_Border"
```
[the remark this answers is not in the document]
Comment (robloo) [reply src/Avalonia.Themes.Fluent/Controls/HeaderedContentControl.xaml]: Ok, no worries. I wanted to be sure I wasn't missing something myself.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0040058-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0040156-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**